lib/rules/complexity.js

```diff
@@ -63,13 +65,10 @@ module.exports = {
         const option = context.options[0];
         let THRESHOLD = 20;
 
-        if (typeof option === "object" && Object.prototype.hasOwnProperty.call(option, "maximum") && typeof option.maximum === "number") {
+        if (typeof option === "object") {
             THRESHOLD = option.maximum;
-        }
-        if (typeof option === "object" && Object.prototype.hasOwnProperty.call(option, "max") && typeof option.max === "number") {
             THRESHOLD = option.max;
```

Comment:
This feels incoherent to me. Maybe I'm reading the diff wrong, but I think the new code looks like this:

```js
if (typeof option === "object") {
    THRESHOLD = option.maximum;
    THRESHOLD = option.max;
}
```

Am I just misreading things? Or can we simplify/change something here?

Reply:
You're right. Maybe we can simplify like this?
```js
THRESHOLD = option.maximum || option.max
```

Or just revert the change? How do you think?

Reply:
I think the simplification is okay- we just need to check that the order is correct. (I vaguely remember that maximum might have been deprecated in favor of max.)